### ISSUE TITLE
DeBruin Forcing has instable memory usage

### ISSUE THREAD

**Issue by Jaapel** [Deltares/hydromt#618]
### HydroMT version checks

- [X] I have checked that this issue has not already been reported.
- [X] I have checked that this bug exists on the latest version of HydroMT.

### Reproducible Example

Wflow build.ini: https://github.com/Deltares/blueearth_cst/blob/1860270b17d06887536e249cbcc4470bd3546264/config/wflow_build_forcing.ini
Being used here: https://github.com/Deltares/blueearth_cst/blob/1860270b17d06887536e249cbcc4470bd3546264/src/downscale_climate_forcing.py

### Current behaviour

I run the downscaling script on a small area in Gabon.
The memory usages spikes up to ~12GB. We tried to add chunks to the data catalog items, but to no avail.

### Desired behaviour

I would like the memory usage to be stable based on the dask workers I supply (where can I supply these?), and based on the chunksize I supply. So if I have 4 workers that have 2 GB memory each and I chunk the source data in 500MB pieces, the workflow will not exceed 8GB memory. This way, I can reliably run larger areas without going OOM.

### Additional context

Related to https://github.com/Deltares/hydromt/pull/28

**Comment by dalmijn** [comment]
@Jaapel I think this is not necessarily the result from the DeBruin method. This has more to do with the scheduling of dask combined with multiple datasets. (which happens in the  `setup_temp_pet_forcing` method in hydromt_wflow)

**Comment by savente93** [comment]
That is true, though I think mentioning DeBruin is more about logging a concrete example of where it actually happens, which is useful, than it is about DeBruin itself

**Comment by Jaapel** [comment]
That is where is happens in the code anyways!

**Comment by savente93** [comment]
Honestly, just thinking about it, this means that this might be a good candidate for a benchmark when we get to that
